storage_devices/disk.py

```diff
@@ -311,4 +324,25 @@ def get_sysfs_addr(cls, device_id):
 
         return sysfs_addr.full_path
 
+    @classmethod
+    def identify(cls, device_name: str) -> bool:
+        output = TestRun.executor.run(
+            f"realpath /sys/block/{device_name}/device/driver | grep virtio"
+        )
+        return output.exit_code == 0
+
 
+class DiskRegistry:
+    driver_types = [NvmeDisk, SataDisk, VirtioDisk]
+
+    @classmethod
+    def get_disk_type(cls, disk_path: str):
+        device_name = TestRun.executor.run(f"realpath {disk_path}").stdout.split("/")[2]
+        recognized_types = [
+            disk_type for disk_type in cls.driver_types if disk_type.identify(device_name)
+        ]
+        if len(recognized_types) == 0:
+            raise TypeError("Framework is not able to recognise disk type")
+        if len(recognized_types) > 1:
+            raise TypeError("One disk got at least 2 disk types")
```

Comment:
Extending the error messages with the path would be quite an improvement for the point of view of future debugging :)

Reply:
In the second case printing all recognised types would be useful as well